doc/pfc_asym/PFC_Asymmetric_HLD.md

```diff
@@ -0,0 +1,391 @@
+
+*Teardown steps*
+
+* Disable asymmetric PFC on all server interfaces
+
+For every server port(setup["server_ports"]) execute command:
+
+```config interface pfc asymmetric [SERVER_PORT] off```
+
+## Test
+
+## Test cases
+
+### Test case # 1 – Asymmetric PFC Off Generate PFC frames
+#### Test objective
+Verify that DUT generates PFC frames only on lossless priorities when asymmetric PFC is disabled
+
+#### Used fixtures
+setup
+
+#### Test steps
+- Setup:
+  - Start ARP responder
+  - Limit maximum bandwith rate on the destination port by setting "1" into SAI port attribute SAI_PORT_ATTR_QOS_SCHEDULER_PROFILE_ID
+  
+- Clear all counters for all ports
+- Get lossless priorities
+- Get lossy priorities
+- Get server ports info
+- Get non server port info (Portchannel peers)
+- Send 10000 packets for lossless priorities from all server ports (src) to non-server port (dst)
+- Verify that some packets are dropped on src ports, which means that Rx queue is full
+- Verify that PFC frames are generated for lossless priorities from src ports
+- Send 10000 packets for lossy priorities from all server ports (src) to non-server port (dst)
+- Verify that PFC frames are not generated for lossy priorities
+
+- Teardown:
+  - Restore maximum bandwith rate on the destination port by setting "0" into SAI port attribute SAI_PORT_ATTR_QOS_SCHEDULER_PROFILE_ID
+   - Stop ARP responder
+
+### Test case # 2 – Asymmetric PFC Off RX pause frames
+#### Test objective
+Verify that while receiving PFC frames DUT drops packets only for lossless priorities (RX and Tx queue buffers are full)
+
+#### Used fixtures
+setup, pfc_storm_runner
+
+#### Test steps
+- Setup:
+  - Start ARP responder
+
+- Clear all counters for all ports
+- Get lossless priorities
+- Get lossy priorities
+- Get server ports info
+- Get non server port info (Portchannel peers)
+- Send 10000 packets for lossy priorities from non-server port (src) to all server ports (dst)
+- Verify that packets are not dropped on src port
+- Verify that packets are not dropped on dst ports
+- Verify that packets are transmitted from from dst ports
+- Send 10000 packets for lossless priorities from non-server port (src) to all server ports (dst)
+- Verify that some packets are dropped on src port, which means that Rx queue is full
+- Verify that some packets are dropped on dst ports, which means that Tx buffer is full
+
+- Teardown:
+  - Stop ARP responder
+
+### Test case # 3 – Asymmetric PFC On Generate PFC frames
+#### Test objective
+Verify that DUT generates PFC frames only on lossless priorities when asymmetric PFC is enabled
+
+#### Used fixtures
+setup, enable_pfc_asym
+
+#### Test steps
+- Setup:
+  - Start ARP responder
+  - Limit maximum bandwith rate on the destination port by setting "1" into SAI port attribute SAI_PORT_ATTR_QOS_SCHEDULER_PROFILE_ID
+
+- As SAI attributes stores PFC values like a bit vector, calculate bitmask for each PFC mode according to configured "lossless" priorities:
```

Comment:
Please explain here why we need it (configuration correctness verification)

Reply:
Fixed